### ISSUE TITLE
Ugly hack in #10557 breaks attach() specification of ClientConnector

### ISSUE THREAD

**Issue by tangomannX** [vaadin/framework#10987]
Since Bugfix #10557 (Vaadin version 8.3.2) Tabsheet calls attach() and detach() when selected tab changes, breaking the attach() [specification](https://vaadin.com/api/com/vaadin/server/ClientConnector.html#attach--) of ClientConnector. 

```
       if (tab != null && !Objects.equals(tab.getComponent(), component) && tab.getComponent() != null && tab.getComponent().isAttached()) {
            tab.getComponent().detach();
            tab.getComponent().attach(); // ugly hack
        }
```
An already attached tab is being detached() an re-attached() again when selected by a user. This breaks the specification of the attach method:

> The caller of this method is Connector#setParent(ClientConnector) if the parent is itself already attached to the session. If not, the parent will call the attach() for all its children when it is attached to the session. This method is always called before the connector's data is sent to the client-side for the first time.

The comments in #10557 already show some users having problems with this unexpected and unspecified behaviour in existing Vaadin applications.
